enumerables.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    i = 0
+    result = []
+    while i < length
+      answer = yield(self[i])
+      result << self[i] unless answer == false
+      i += 1
+    end
+    result
+  end
+
+  def my_all?(pattern = nil)
+    if block_given?
+      my_each { |item| return unless yield item }
+    elsif pattern
+      my_each { |item| result & pattern == item || pattern == item.class }
+    else
+      my_each { |item| result & item }
+    end
+    true
+  end
```

Comment:
Remember that `my_all?` is supposed to return either `true` or `false`. However, line 41 returns `nil`.

Also, did you consider when the user passes a proc as an argument?

Finally, where did you define the `result` you are using on lines 43 and 45? It is blowing up the method execution, thus I can't test the method properly.